### PR TITLE
Refine types + rework handler a bit + bump python to 3.10

### PR DESCRIPTION
**Pull request by edisontim**

### CODE REVIEW

File: overlore/db/handler.py [hunk 1]
```diff
@@ -195,39 +188,46 @@ def init(self, path: str = "./events.db") -> DatabaseHandler:
         self.realms = Realms.instance().init()
         return self
 
-    def get_by_id(self, event_id: int) -> Any:
+    def get_by_id(self, event_id: int) -> StoredEvent:
+        # lock mutex
         self.__lock()
```
Comment (jobez) [inline overlore/db/handler.py:193]: do reads need to be thread safe? a quick google implies that theres only one writer permitted, but reads are threadsafe

File: overlore/db/handler.py [hunk 2]
```diff
@@ -195,39 +188,46 @@ def init(self, path: str = "./events.db") -> DatabaseHandler:
...
+        records = [[*list(tup[:5]), (tup[5], tup[6]), (tup[7], tup[8])] for tup in records]
         return records
 
-    def process_event(self, event: dict[str, Any]) -> int:
-        if self.__get_event_type(event) == EventKeyHash.COMBAT_OUTCOME.value:
-            parsed_event = self.__parse_combat_outcome_event(event)
+    def process_event(self, event: ToriiEvent) -> int:
+        event_emitted = event.get("eventEmitted")
+        if not event_emitted:
+            raise RuntimeError("eventEmitted no present in event")
```
Comment (jobez) [inline overlore/db/handler.py:222]: hmm, thinking about whether we want the whole program to crash when events arent in the shape we expect. i think in the stage we are at now, this makes sense, but potentially in the future we would be good just logging these errors and keeping the system running

**Review by jobez — COMMENTED** [review]
just a sanity check if we need locks for sqlite reads or not. and  can you give a brief description of the changes made to the github workflow stuff? seems like you simplified that matrix stuff because we only care about it running in a particular os?

File: overlore/db/handler.py (hunk 1 above)
Comment (edisontim) [reply overlore/db/handler.py:193]: I'm not sure about reads but if it says that it's thread-safe then it should be fine. I was afraid that there would be race conditions given that we have asynchronous tasks write/reading in the db that's why I added the locks

**Comment by edisontim** [comment]
> just a sanity check if we need locks for sqlite reads or not. and can you give a brief description of the changes made to the github workflow stuff? seems like you simplified that matrix stuff because we only care about it running in a particular os?

I removed the files that were being cached. Honestly it's because I couldn't get the thing to work. Something about the shell not being interactive and not sourcing the .bashrc.

File: overlore/db/handler.py (hunk 1 above)
Comment (jobez) [reply overlore/db/handler.py:193]: skimming thru this: https://ricardoanderegg.com/posts/python-sqlite-thread-safety/, i think we can simplify the locks. im fine with doing this in a follow up pr

**Review by jobez — APPROVED** [review]

**Pull request merged**